frontend/src/composables/useShootEditor/index.js

```diff
@@ -257,33 +312,24 @@ export function useShootEditor (initialValue, options = {}) {
     setEditorValue(shootItem.value)
   }
 
-  function refreshEditor () {
-    if (cm.value) {
-      cm.value.refresh()
-    }
-  }
-
   function reloadEditor () {
     resetEditor()
-    refreshEditor()
   }
 
   function focusEditor () {
-    if (cm.value) {
-      cm.value.focus()
-    }
-  }
-
-  function getEditorScrollInfo () {
-    if (cm.value) {
-      return cm.value.getScrollInfo()
+    if (cmView.value) {
+      cmView.value.focus()
     }
   }
 
   function clearDocumentHistory () {
-    if (cm.value) {
-      cm.value.doc.clearHistory()
-      generation.value = cm.value.doc.changeGeneration()
+    if (cmView.value) {
+      cmView.value.dispatch({
+        effects: historyCompartment.reconfigure([]),
+      })
+      cmView.value.dispatch({
+        effects: historyCompartment.reconfigure(history()),
+      })
```

Comment:
could you kindly explain this code?

Reply:
This code resets the internal history plugin. AFAIK there is no better way to do this. ChatGPT tells you this:
>In CodeMirror 6, the history is managed internally by the history extension, and there isn't a public API to directly reset or clear the history state. Directly manipulating internal state fields or extensions is generally not recommended, as it can lead to unexpected behavior and may break with future updates.

However, simply reconfiguring with
```
cmView.value.dispatch({
  effects: historyCompartment.reconfigure(history()),
})
```

does not reset the history. Explicitly removing and re-adding the history plugin is the only way I got it working.